hdmi/tinyaudio_hw.c

```diff
@@ -419,7 +425,46 @@ static int out_set_parameters(struct audio_stream *stream, const char *kvpairs)
     ALOGV("%s exit",__func__);
     return 0;
 }
+static int parse_hdmi_device_number()
+{
+    struct mixer *mixer;
+    int card = 0;
+    struct mixer_ctl *ctl;
+    enum mixer_ctl_type mixer_type;
+    unsigned int num_values;
+    unsigned int i,id;
+    bool device_status;
 
+    ALOGV("%s enter",__func__);
+    card = get_card_number_by_name("PCH"); 
+    mixer = mixer_open(card);
+    if (!mixer) {
+        ALOGE(" Failed to open mixer\n");
+        return -1;
+    }
+    for(i=0; i < MAX_HDMI_DEVICES; i++ ) {
+       char ctl_name[100] = "HDMI/DP,pcm=";
+       char buffer[10]; 
+       sprintf(buffer, "%d", i);
+       strcat(ctl_name,buffer); 
+       strcat(ctl_name," Jack");
+       ALOGV("ctl_name %s",ctl_name); 
+       ctl = mixer_get_ctl_by_name(mixer, ctl_name);
+       if(ctl) {
+           mixer_type = mixer_ctl_get_type(ctl);
+           num_values = mixer_ctl_get_num_values(ctl);
+           device_status = mixer_ctl_get_value(ctl, 1);
+           ALOGV(" mixer_ctl_get_value %s", mixer_ctl_get_value(ctl, 1) ? "On" : "Off");  
+       }
+       if(device_status == 1) {
+           ALOGV("status of device %d is ON",i);
+           ALOGV("%s exit",__func__);
+           return i;
+       }   
+    }
+    ALOGV("%s exit",__func__);
+    return DEFAULT_DEVICE;
```

Comment:
Would it be better to return -1 from here, and hardcode the default from the caller ?